src/main/java/br/eletra/frontend/controllers/ScreenController.java

```diff
@@ -20,43 +31,73 @@ public class ScreenController {
     @FXML
     private Accordion accordion;
 
-    private final LinesInitializer linesInitializer = new LinesInitializer();
-    private final ModelsInitializer modelsInitializer = new ModelsInitializer();
+    private final Session session = HibernateConfiguration.getSessionFactory().openSession();
+    private LineDAO lineDAO;
+    private CategoryDAO categoryDAO;
+    private ModelDAO modelDAO;
+
+    public ScreenController() {
+    }
 
-    public void initialize() {
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
         titledPaneModels.setDisable(true);
 
-        comboBoxLines.setItems(linesInitializer.lines);
-        comboBoxLines.setOnAction(event -> handleLineSelection());
+        try {
+            lineDAO = new LineDAO(session);
+            categoryDAO = new CategoryDAO(session);
+            modelDAO = new ModelDAO(session);
+
+            List<LineEntity> lines = lineDAO.getAllLines();
+            ObservableList<String> lineNames = FXCollections.observableArrayList();
+            for (LineEntity line : lines) {
+                lineNames.add(line.getName());
+            }
+            comboBoxLines.setItems(lineNames);
+
+            comboBoxLines.setOnAction(event -> handleLineSelection());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
     }
 
     private void handleLineSelection() {
         String selectedLine = comboBoxLines.getValue();
         if (selectedLine != null) {
             titledPaneModels.setDisable(false);
             displayModels(selectedLine);
+            expandModels();
         }
     }
 
+
     private void displayModels(String selectedLine) {
-        Map<String, List<String>> subcategories = modelsInitializer.initializeModels().get(selectedLine);
-        if (subcategories != null) {
-            TreeItem<String> rootItem = new TreeItem<>(selectedLine);
-            for (String subcategory : subcategories.keySet()) {
-                TreeItem<String> subcategoryItem = new TreeItem<>(subcategory);
-                List<String> models = subcategories.get(subcategory);
-                for (String model : models) {
-                    subcategoryItem.getChildren().add(new TreeItem<>(model));
+        try {
+            LineEntity line = lineDAO.getLineByName(selectedLine);
+            if (line != null) {
```

Comment:
Essa verificação é inútil, visto que ela já ocorreu na linha 66, por favor remover.